README.md

```diff
@@ -1,73 +1,122 @@
+# 🏗 Scaffold-ETH 2 PWA 📱
 
-⚙️ Built using NextJS, RainbowKit, Hardhat, Wagmi, and Typescript.
+This **forkable** project provides the infraestructure to build a Progressive Web App (PWA) using Scaffold-ETH 2 base features, plus PWA oriented extra features, like Push Notifications and the capability to Install the PWA on your device.
 
-- ✅ **Contract Hot Reload**: Your frontend auto-adapts to your smart contract as you edit it.
-- 🔥 **Burner Wallet & Local Faucet**: Quickly test your application with a burner wallet and local faucet.
-- 🔐 **Integration with Wallet Providers**: Connect to different wallet providers and interact with the Ethereum network.
+To learn more about base Scaffold-ETH 2 features and development guide, check our [Docs](https://docs.scaffoldeth.io/) and [Website](https://scaffoldeth.io/).
 
-![Debug Contracts tab](https://github.com/scaffold-eth/scaffold-eth-2/assets/55535804/1171422a-0ce4-4203-bcd4-d2d1941d198b)
+## 🏃 Quick Start
 
-## Requirements
+To get started with Scaffold-ETH 2 PWA, follow the steps below:
 
-Before you begin, you need to install the following tools:
+### 1 . Clone this repo & install dependencies
 
-- [Node (v18 LTS)](https://nodejs.org/en/download/)
-- Yarn ([v1](https://classic.yarnpkg.com/en/docs/install/) or [v2+](https://yarnpkg.com/getting-started/install))
-- [Git](https://git-scm.com/downloads)
+```bash
+gh repo clone BuidlGuidl/PWA-burner-wallet
+cd PWA-burner-wallet
+yarn install
+```
 
-## Quickstart
+### 2 . Setting up Firebase
 
-To get started with Scaffold-ETH 2, follow the steps below:
+> Note : You can also use other database as well, we are using Firebase for this example checkout `packages/nextjs/database/firebase`.
 
-1. Clone this repo & install dependencies
+Copy `packages/nextjs/.env.example` into `packages/nextjs/.env` file and fill in your Firebase credentials which starts with `FB`.
 
-```
-git clone https://github.com/scaffold-eth/scaffold-eth-2.git
-cd scaffold-eth-2
-yarn install
+Checkout [this article](https://softauthor.com/add-firebase-to-javascript-web-app/#add-firebase-sdk-to-javascript-web-app) on how to get config variables from Firebase.
+
+> Hint: If it's your first time using Firebase, please read the article from the beginning, as you'll need to create a Firebase project and register a web app first.
+
+We use the Firebase config variables in our `packages/nextjs/database/firebase/config` file:
+
+```js
+// Your web app's Firebase configuration
+const firebaseConfig = {
+  apiKey: process.env.FB_API_KEY,
+  authDomain: process.env.FB_AUTH_DOMAIN,
+  projectId: process.env.FB_PROJECT_ID,
+  storageBucket: process.env.FB_STRG_BUCKET,
+  messagingSenderId: process.env.FB_MESSAGING_SENDER_ID,
+  appId: process.env.FB_APP_ID,
+};
 ```
 
-2. Run a local network in the first terminal:
+> Hint : Make sure you have read and write [rules](https://firebase.google.com/docs/firestore/security/get-started#testing_rules) allowed in Firebase.
```

Comment:
I think we need to be more clear, that you need to set up a Firestore.

Firestore != Firebase, and creating a Firestore wasn't mentioned before.

We could also be more explicit / copy the Firestore rules here. Maybe only allow read/write for all in the `subscriptions` collection?

Reply:
> I think we need to be more clear, that you need to set up a Firestore.

ohh yes we need to tell about [creating  firestore database](https://firebase.google.com/docs/firestore/quickstart)

> We could also be more explicit / copy the Firestore rules here. Maybe only allow read/write for all in the subscriptions collection?

Yup, @Pabl0cks please could you update this 🙌, here is the rule(tested it and works) : 

```
rules_version = '2';

service cloud.firestore {
  match /databases/{database}/documents {

    // This rule allows anyone with your Firestore database reference to view, edit,
    // and delete all data in your Firestore subscriptions collection
    match /subscriptions/{subscription} {
      allow read, write: if true;
    }
  }
}
```

Reply:
I've just pushed in https://github.com/BuidlGuidl/PWA-burner-wallet/pull/1/commits/4af299ac1d8abcd727f7524e546a89b49cb5d9f1 the updated version for **2 . Setting up Firebase** section. 

Let me know if you would change anything else there ♥

Reply:
@technophile-04 I saw you added this:

```
// This rule allows anyone with your Firestore database reference to view, edit,
// and delete all data in your Firestore subscriptions collection
```

Maybe we can move that warning out of the code block? It seems that it's not the ideal setup (for security reasons)

I don't understand the full picture yet: with that rule, who is able to write? Everyone in the front-end? Or just from a backed that have the secret? Maybe it's just because we have the "notify all" option in the testing app.

We could also think a better example. In any case, I think it's important to warn people about that setting.

thanks!

Reply:
> I don't understand the full picture yet: with that rule, who is able to write? Everyone in the front-end? Or just from a backed that have the secret? Maybe it's just because we have the "notify all" option in the testing app.

I actually copied and pasted those lines from when you selected Firebase test mode, but after doing some research 

It seems that if anyone has your "project-id" and "webiapi-key" then can just make API request to firestore REST API : 

```
curl -X DELETET https://firestore.googleapis.com/v1/projects/{PROJECT_ID}/databases/(default)/documents/subscriptions/{DOCUMENT_ID}?key={web-api-key}
```

But this is not the case with us since we not are not leaking PROJECT_ID, WEB_API_KEY because we are accessing Firebase through NEXTJs serverless functions 

If we would have used Firebase on client side stored those variable in `NEXT_PUBLIC` then people can get those keys and make a malicious request if the rules were not set properly

Reply:
> because we are accessing Firebase through NEXTJs serverless functions

Great!! Thank you :)